docs/overrides.md

```diff
@@ -0,0 +1,212 @@
+# Modifying the default BKPR configuration
+
+BKPR uses [jsonnet](https://jsonnet.org/) to describe Kubernetes manifests. Jsonnet is a simple programming language that generates JSON. If you are new to jsonnet and want to extend BKPR to suit your particular use case, we recommend to go through the [jsonnet basic tutorial](https://jsonnet.org/learning/tutorial.html) first, in order to get familiar with the syntax.
+
+When deploying BKPR with `kubeprod install` two files get generated in your local folder::`kubeprod-autogen.json`, with your cloud seetings, and `kubeprod-manifest.jsonnet` with the root manifest for the Kubernetes objectsts.
+
+To modify the default configuration you will need to edit the `kubeprod-manifest.jsonnet` manifest and add a set of jsonnet "overrides". In most cases, we will be using the jsonnet `+` operator, which will merge the JSON after that `+` operator into the parent object.
+
+When BKPR is first deployed in a GKE cluster the following `kubeprod-manifest.jsonnet` gets generated:
+
+```
+// Cluster-specific configuration
+(import "https://releases.kubeprod.io/files/v1.1.1/manifests/platforms/gke.jsonnet") {
+	config:: import "kubeprod-autogen.json",
+	// Place your overrides here
+}
+```
+
+This snip basically imports the default root manifest for your platform (in this example, GKE) and applies the generated cloud specific configuration.
+
+If you want to check what manifests this default configuration would produce, you can use [`kubecfg`](https://github.com/ksonnet/kubecfg) a tool to manage Kubernetes resources as code and that understands `jsonnet` (actually, `kubeprod` uses `kubecfg` libraries internally):
+
+```
+kubecfg show kubeprod-manifest.jsonnet
+```
+
+## Example 1: Modifying Maximum Number of Replicas for `oauth2-proxy`
+
+This example modifies the maximum number of replicas for the `oauth2-proxy` Deployment.
+
+This is a snippet of the YAML generated by `kubecfg show`, specifically the one for the `oauth2-proxy` HorizontalPodAutoscaler, which will scale the deployment based on load:
+
+```
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+ metadata:
+   labels:
+     name: oauth2-proxy
+   name: oauth2-proxy
+   namespace: kubeprod
+ spec:
+   maxReplicas: 10
+   minReplicas: 1
+   scaleTargetRef:
+     apiVersion: apps/v1beta1
+     kind: Deployment
+     name: oauth2-proxy
+```
+
+That snippet comes from the [`oauth2-proxy.jsonnet`](https://github.com/bitnami/kube-prod-runtime/blob/master/manifests/components/oauth2-proxy.jsonnet) file, that gets imported in the previous one. Specifically, this is the part about the HorizontalPodAutoscaler:
```

Comment:
Is it worth starting from the platform file imported in `kubeprod-manifest.jsonnet`, and show how to walk down the imports to the oauth2-proxy file?  Going "bottom up" (just jumping to "knowing" that the oauth2 config is in this file) seems a bit magical to me otherwise.